File: 350 Intersection of Two Arrays II.java
class Solution {
    public int[] intersect(int[] nums1, int[] nums2) {
        int m = nums1.length; 
        int n = nums2.length;
        if (m == 0 || n == 0) {
            return (new int[0]);
        }

         Arrays.sort(nums1);
         Arrays.sort(nums2);
         List<Integer> res = new ArrayList<Integer>();

         int p = 0, q = 0;
         while (true) {
             if (p == m || q == n) break;

             if (nums1[p] < nums2[q]) {
                 p++;
             }
             else if (nums1[p] > nums2[q]) {
                 q++;
             }
             else {
                res.add(nums1[p]);
                p++; q++;
             }
         }

         int[] resa = new int[res.size()];
         for (int i = 0; i < resa.length; i++) {
             resa[i] = res.get(i);
         }
         return resa;        

    }
}

File: 350 Intersection of Two Arrays II.go
func intersect(nums1 []int, nums2 []int) []int {
    if len(nums1) > len(nums2) { return intersect(nums2, nums1) }
    sort.Ints(nums1)
    sort.Ints(nums2)
    res := make([]int, 0, len(nums1))
    p1 := 0
    p2 := 0
    for p1 < len(nums1) && p2 < len(nums2) {
        if nums1[p1] == nums2[p2] {
            res = append(res, nums1[p1])
            p1++
            p2++
        } else if nums1[p1] > nums2[p2] {
            p2++
        } else {
            p1++
        }
    }
    return res

}
